packages/react-native-editor/bin/extract-used-strings.js

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env node
+/**
+ *
+ * Extracts used strings from source-map files to a JSON file.
+ *
+ * Usage:
+ * 	- Extract used strings referenced in Gutenberg:
+ * 	  node extract-used-strings used-strings.json
+ *
+ * 	- Extract used strings referenced in Gutenberg and other plugins with their own domain:
+ * 	  node extract-used-strings used-strings.json domain-plugin-1 <PLUGIN-1_SOURCE_PATH> domain-plugin-2 <PLUGIN-2_SOURCE_PATH>
+ *
+ * The format of the JSON file is:
+ * {
+ * 	"gutengerg": {
+ * 	  "<string>": {
+ * 	    "string": String value.
+ * 		"stringPlural": String value with its plural form. [optional]
+ * 		"comments": Comments for translators. [default value is an empty string]
+ * 		"reference": Array containing the paths of the source files that reference the string.
+ * 		"platforms": Array containing the platforms where the string is being used, values are "android" | "ios" | "web".
+ * 	  },
+ * 	  ...
+ * 	},
+ * 	"domain-plugin-1": {
+ * 	  ...
+ * 	},
+ * 	...
+ * }
+ *
+ */
+
+/**
+ * External dependencies
+ */
+const gettextParser = require( 'gettext-parser' );
+const fs = require( 'fs' );
+const path = require( 'path' );
+const childProcess = require( 'child_process' );
+
+const getStringsFromPotFile = ( potFileName ) => {
+	const potData = fs.readFileSync( potFileName );
+	const po = gettextParser.po.parse( potData );
+
+	const getStringsFromContext = ( contextTranslations ) =>
+		Object.values( contextTranslations ).map(
+			( {
+				msgid: stringSingular,
+				msgid_plural: stringPlural,
+				msgctxt,
+				comments,
+			} ) => ( {
+				string: msgctxt
+					? `${ msgctxt }\u0004${ stringSingular }`
+					: stringSingular,
+				stringPlural:
+					msgctxt && stringPlural
+						? `${ msgctxt }\u0004${ stringPlural }`
+						: stringPlural,
+				context: msgctxt,
+				comments: comments?.extracted || '',
+				reference: comments?.reference.split( '\n' ) || [],
+			} )
+		);
+
+	return Object.keys( po.translations ).reduce( ( result, context ) => {
+		const items = getStringsFromContext( po.translations[ context ] );
+		return [ ...result, ...items ];
+	}, [] );
+};
+
+const createString = ( previousString, string, platform ) =>
+	previousString
+		? {
+				...previousString,
```

Comment:
Is the indentation off here?

Reply:
I reformatted the file and looks like the indentation is correct, is producing a different result on your side?

Reply:
OK.  Seems like there is too many tabs in there but it's probably just the ternary formatting that pushes out `...previousString` , Thanks for checking!